fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/Availability.java

```diff
@@ -22,33 +22,21 @@
      *
      * @return A set of names for datasources backing this table
      */
-    SortedSet<TableName> getDataSourceNames();
-
-    /**
-     * The availability for a given column.
-     *
-     * @param c  A column
-     *
-     * @return The list of intervals that column is available for.
-     */
-    List<Interval> get(Column c);
+    Set<TableName> getDataSourceNames();
 
     /**
      * The availability of all columns.
      *
      * @return The intervals, by column, available.
      */
-    Map<Column, List<Interval>> getAvailableIntervals();
+    Map<Column, Set<Interval>> getAllAvailableIntervals();
```

Comment:
It's a List because it makes collapsing the intervals down _much_ more performant. It's the same general idea that searching through an ordered collection is _much_ faster than an unordered collection (which essentially requires a scan). @michael-mclawhorn can go deeper on this if you'd like 😉